course05/homework02/freekpaans/linked_list.h

```diff
@@ -0,0 +1,78 @@
+#include <cstddef>
+#include <string>
+#include <memory>
+#include <iterator>
+
+class LinkedList{
+public:
+	using size_type = size_t;
+
+	struct Link
+	{
+		std::string mValue;
+		std::unique_ptr<Link> mNext;
+	};
+
+	using LinkPtr = std::unique_ptr<Link>;
```

Comment:
Is there a way to use the alias also in the struct definition (`LinkPtr mNext`)? There's a bit of a dependency problem here.

Reply:
You can forward declare it

```
struct A
{
    struct B;
    using BPtr = std::unique_ptr<B>;

    struct B
    {
        BPtr mNextB;
    };

    B mB;
};
```